PecanProject/sipnet
# Event Handler Infrastructure

This PR adds infrastructure to SIPNET to enable handling of agronomic events.

## Structure
* Use a compile-time flag to turn this functionality on/off
* Put this flag (and similar future flags) in a separate file to allow for easier testing (by having only one file that needs to be replaced to control tested functionality, and also to have nothing else in that file)
* Create unit test(s)
* Update the github workflow to run the new tests

## Design 
* Read event information from a specified file, one line per event.
* The spec for the agronomic events is [here](https://github.com/ccmmf/modeling_framework/blob/main/sipnet-model-description/input-and-output-parameters.md#agronomic-events); a reviewer with access to that doc should make sure this code follows that spec.
* Store event information in a similar fashion as climate data
* Future (next step, really): augment `sipnet.c` to fetch the appropriate events at a given location in the now-empty `setupEvents` function

## Testing
* This is an attempt (first try?) at adding some infrastructure for unit testing. 
* Basic idea:
  * Unit tests go in a subdirectory of `/tests/sipnet` (or another sub of `/tests/`, if we add specific testing for, say, `estimate`)
  * Create a `model_structures.h` file with the correct compile switches for the functionality to be tested
  * Copy `/tests/Makefile.template` and add the test c-files to it
  * ...and write the tests, of course. The infrastructure should handle the rest.
  * See `tests/sipnet/test_events` for an example

## Notes
* `frontend.c`: it was tempting to simplify this code by removing the #if checks for EVENT_
HANDLER, but that seemed out of character for SIPNET
* `sipnet.c`: I commented out some variable declarations to clean up compiler warnings
